packages/transactions/tests/builder.test.ts

```diff
@@ -267,6 +267,90 @@ test('Make Multi-Sig STX token transfer', async () => {
   expect(serializedSignedTx.toString('hex')).toBe(signedTx);
 });
 
+test('Make Multi-Sig STX token transfer with two transaction signers', async () => {
+  const recipient = standardPrincipalCV('SP3FGQ8Z7JY9BWYZ5WM53E0M9NK7WHJF0691NZ159');
+  const amount = 2500000;
+  const fee = 0;
+  const nonce = 0;
+  const memo = 'test memo';
+
+  const authType = AuthType.Standard;
+  const addressHashMode = AddressHashMode.SerializeP2SH;
+
+  const privKeyStrings = [
+    '6d430bb91222408e7706c9001cfaeb91b08c2be6d5ac95779ab52c6b431950e001',
+    '2a584d899fed1d24e26b524f202763c8ab30260167429f157f1c119f550fa6af01',
+    'd5200dee706ee53ae98a03fba6cf4fdcc5084c30cfa9e1b3462dcdeaa3e0f1d201',
+  ];
+  const privKeys = privKeyStrings.map(createStacksPrivateKey);
+
+  const pubKeys = privKeyStrings.map(pubKeyfromPrivKey);
+  const pubKeyStrings = pubKeys.map(publicKeyToString);
+
+  const transaction = await makeUnsignedSTXTokenTransfer({
+    recipient,
+    amount,
+    fee,
+    nonce,
+    memo: memo,
+    numSignatures: 2,
+    publicKeys: pubKeyStrings,
+    anchorMode: AnchorMode.Any
+  });
+
+  const serializedTx = transaction.serialize();
+
+  const tx =
+    '00000000010401a23ea89d6529ac48ac766f720e480beec7f1927300000000000000000000000000000000' +
+    '000000000002030200000000000516df0ba3e79792be7be5e50a370289accfc8c9e03200000000002625a0' +
+    '74657374206d656d6f00000000000000000000000000000000000000000000000000';
+
+  expect(serializedTx.toString('hex')).toBe(tx);
+
+  const bufferReader = new BufferReader(serializedTx);
+  const deserializedTx = deserializeTransaction(bufferReader);
+
+  expect(deserializedTx.auth.authType).toBe(authType);
+
+  expect(deserializedTx.auth.spendingCondition!.hashMode).toBe(addressHashMode);
+  expect(deserializedTx.auth.spendingCondition!.nonce.toString()).toBe(nonce.toString());
+  expect(deserializedTx.auth.spendingCondition!.fee.toString()).toBe(fee.toString());
+  expect(deserializedTx.auth.spendingCondition!.signer).toEqual(
+    'a23ea89d6529ac48ac766f720e480beec7f19273'
+  );
+  const deserializedPayload = deserializedTx.payload as TokenTransferPayload;
+  expect(deserializedPayload.amount.toString()).toBe(amount.toString());
+
+  const signer = new TransactionSigner(deserializedTx);
+  // sign once
+  signer.signOrigin(privKeys[0]);
+
+  // serialize
+  const partiallySignedSerialized = deserializedTx.serialize();
+
+  // deserialize
+  const bufferReader2 = new BufferReader(partiallySignedSerialized);
+  const partiallySigned = deserializeTransaction(bufferReader2);
+
+  // finish signing with new TransactionSigner
+  const signer2 = new TransactionSigner(partiallySigned);
+  signer2.signOrigin(privKeys[1]);
+  signer2.appendOrigin(pubKeys[2]);
+
+  const serializedSignedTx = partiallySigned.serialize();
+
+  const signedTx =
```

Comment:
is there some other way to test the success of signing other than checking the serialized output?

I'm guessing you got this output by just printing the result that you got, so none of us human readers can know if this is really expected or not.

Reply:
I think it's helpful to check output blob, but agreed with @gregorycoppola, would be helpful if individual properties relevant to signing could be explicitly checked (I think you could test the public keys, ordering, and count?)

Reply:
+1 -- the test should check both against a blob, and against the reconstructed object properties.